public/index.html

```diff
@@ -13,27 +13,44 @@
         j.src = 'https://www.googletagmanager.com/gtm.js?id=' + i + dl + auth;
         f.parentNode.insertBefore(j, f);
       }
+      function gtmConfig(userId, internalUser, tagId, auth) {
+        window.dataLayer = window.dataLayer || [];
+        if (userId) {
+          window.dataLayer.push({ user_id: userId });
+        }
+        if (internalUser) {
+          window.dataLayer.push({ internal_user: internalUser });
+        }
+        gtm(window, document, 'script', 'dataLayer', tagId, auth);
+      }
       if (window.location.hostname.match(/^(www.)?staging.terraware.io$/i)) {
-        gtm(
-          window,
-          document,
-          'script',
-          'dataLayer',
-          'GTM-TNX8NWM',
-          '&gtm_auth=Iwp9NkwNGO7TelQuq94FhQ&gtm_preview=env-7&gtm_cookies_win=x'
-        );
+        window.INIT_GTAG = function (userId, internalUser) {
+          gtmConfig(
+            userId,
+            internalUser,
+            'GTM-TNX8NWM',
+            '&gtm_auth=Iwp9NkwNGO7TelQuq94FhQ&gtm_preview=env-7&gtm_cookies_win=x'
+          );
+        };
       } else if (window.location.hostname.match(/^(www.)?terraware.io$/i)) {
-        gtm(
-          window,
-          document,
-          'script',
-          'dataLayer',
-          'GTM-TNX8NWM',
-          '&gtm_auth=oxl4mxgvnvET8ov_75EuYA&gtm_preview=env-1&gtm_cookies_win=x'
-        );
+        window.INIT_GTAG = function (userId, internalUser) {
+          gtmConfig(
+            userId,
+            internalUser,
+            'GTM-TNX8NWM',
+            '&gtm_auth=oxl4mxgvnvET8ov_75EuYA&gtm_preview=env-1&gtm_cookies_win=x'
+          );
+        };
       } else {
-        // Uncomment this line to track non-staging/prod environments
-        // gtm(window, document, 'script', 'dataLayer', 'GTM-TNX8NWM', '&gtm_auth=1591mjZEVL_pulZkOX4gWg&gtm_preview=env-11&gtm_cookies_win=x');
+        // Uncomment this function to track non-staging/prod environments
+        window.INIT_GTAG = function (userId, internalUser) {
```

Comment:
this will be commented out prior to checking-in, for new we need this to test in vercel